portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -22,11 +31,83 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
+public class DataServlet extends HttpServlet{
+    /**
+    * set of constants that are POST request parameter keys and Entity property keys
+    */
+    private static final String CONTACT = "Contact";
+    private static final String FIRST_NAME = "firstname";
+    private static final String LAST_NAME = "lastname";
+    private static final String EMAIL = "email";
+    private static final String SUBJECT = "subject";
+    private static final String TIMESTAMP = "timestamp";
+    private static final Gson gson = new Gson();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query(CONTACT);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        int displayTasks = Integer.parseInt(request.getParameter("displayComments"));
+        List<Contact> contacts = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            //Display x amounts of tasks
+            if (displayTasks == 0){
+                break;
+            }else{
+                displayTasks--;
+            }
```

Comment:
Instead of mutating displayTasks, before adding to the list of contacts, checkif the size of the arraylist == displayTasks